src/friendly-iframe-embed.js

```diff
@@ -452,6 +454,144 @@ export class FriendlyIframeEmbed {
       this.visibilityObservable_.fire(this.visible_);
     }
   }
+
+  /**
+   * @return {!HTMLBodyElement}
+   * @visibleForTesting
+   */
+  getBodyElement() {
+    return /** @type {!HTMLBodyElement} */ (
+        (this.iframe.contentDocument || this.iframe.contentWindow.document)
+            .body);
+  }
+
+  /**
+   * @return {!./service/vsync-impl.Vsync}
+   * @visibleForTesting
+   */
+  getVsync() {
+    return vsyncFor(this.win);
```

Comment:
Thanks @alanorozco again for sharing the experience. We have started to talk about how we can improve tests in AMP in general.

@rsimha-amp @choumx @keithwrightbos @taymonbeal 
FYI this is the missing infrastructure that I mentioned in the meeting for unit test in AMP.